src/TokenType.h

```diff
@@ -0,0 +1,46 @@
+#ifndef TOKEN_TYPE_H
+#include <string>
+#define TOKEN_TYPE_H
+enum TokenType {
+  // Single-character tokens.
+  LEFT_PAREN, RIGHT_PAREN, LEFT_BRACE, RIGHT_BRACE,
+  COMMA, DOT, MINUS, PLUS, SEMICOLON, SLASH, STAR,
+
+  // One or two character tokens.
+  BANG, BANG_EQUAL,
+  EQUAL, EQUAL_EQUAL,
+  GREATER, GREATER_EQUAL,
+  LESS, LESS_EQUAL,
+
+  // Literals.
+  IDENTIFIER, STRING, NUMBER,
+
+  // Keywords.
+  AND, CLASS, ELSE, FALSE, FUN, FOR, IF, NIL, OR,
+  PRINT, RETURN, SUPER, THIS, TRUE, VAR, WHILE,
+
+  EOF
+};
+
+std::string TokenString[] = {
+  // Single-character tokens.
+  "LEFT_PAREN", "RIGHT_PAREN", "LEFT_BRACE", "RIGHT_BRACE",
+  "COMMA", "DOT", "MINUS", "PLUS", "SEMICOLON", "SLASH", "STAR",
+
+  // One or two character tokens.
+  "BANG", "BANG_EQUAL",
+  "EQUAL", "EQUAL_EQUAL",
+  "GREATER", "GREATER_EQUAL",
+  "LESS", "LESS_EQUAL",
+
+  // Literals.
+  "IDENTIFIER", "STRING", "NUMBER",
+
+  // Keywords.
+  "AND", "CLASS", "ELSE", "FALSE", "FUN", "FOR", "IF", "NIL", "OR",
+  "PRINT", "RETURN", "SUPER", "THIS", "TRUE", "VAR", "WHILE",
+
+  "EOF"
+};
```

Comment:
maybe then this can be a function returning a string after a lookup on a  `static const std::map<Tokentype, string>`

Reply:
also thought about the possibility for this to be a `map`.  What's the best approach or must commonly used for similar situations ?

Reply:
maybe smth like:

```
 const std::string TokenTypeString(const TokenType val)  {
  static const std::map<TokenType, std::string> tokenTypeString{
       .....
      {TokenType::EOF, "EOF"},
      .....
      return tokenTypeString.find(val) -> second;
```